tests/components/esphome/test_config_flow.py

```diff
@@ -0,0 +1,159 @@
+"""Test config flow."""
+from collections import namedtuple
+from unittest.mock import patch, MagicMock
+
+import pytest
+
+from homeassistant.components.esphome import config_flow
+from tests.common import mock_coro
+
+MockDeviceInfo = namedtuple("DeviceInfo", ["uses_password", "name"])
+
+
+@pytest.fixture(autouse=True)
+def aioesphomeapi_mock():
+    """Mock aioesphomeapi."""
+    with patch.dict('sys.modules', {
+        'aioesphomeapi': MagicMock(),
+    }):
+        yield
+
+
+@pytest.fixture
+def mock_client():
+    """Mock APIClient."""
+    with patch('aioesphomeapi.APIClient') as mock_client:
+        def mock_constructor(loop, host, port, password):
+            """Fake the client constructor."""
+            mock_client.host = host
+            mock_client.port = port
+            mock_client.password = password
+            return mock_client
+
+        mock_client.side_effect = mock_constructor
+        mock_client.start.return_value = mock_coro()
+        mock_client.connect.return_value = mock_coro()
+        mock_client.stop.return_value = mock_coro()
+        mock_client.login.return_value = mock_coro()
+
+        yield mock_client
+
+
+@pytest.fixture(autouse=True)
```

Comment:
Should it be auto used?

Reply:
Otherwise the `try:` `except APIConnectionError` clause will fail because `MagicMock` doesn't inherit from `BaseException`